### DISCUSSION TITLE
Using the GH Api how do I tell when a PR was converted from draft and ready for review

### DISCUSSION THREAD

**Discussion by StingyJack** [community/community#24734] [Apps, API and Webhooks]
<p>Its not going to be the same as CreatedAt when the PR is created as a draft.  UpdatedAt will change if anything else is changed about the PR.  Is there a reliable way to find the date time of when the PR was marked as ready for Review?</p>
<p>When trying to figure out the average time to close across PR’s, using CreatedAt isnt going to produce accurate values.</p>

**Comment by shrink** [comment, marked as answer]
<p>A Pull Request can switch back and forth between draft and ready to review so there’s no single value that you can rely on. However, because a Pull Request is a type of Issue you could derive this information from the Events history of the Pull Request.</p>
<p>The Timeline API is currently in preview so it is subject to change <em>but</em> if you’re comfortable using it during the preview period you could pull the Event history for each Pull Request and find the relevant <code>convert_to_draft</code> or <code>ready_for_review</code> event to determine how you categorise the Pull Request.</p>
<p><a href="https://docs.github.com/en/rest/reference/issues#timeline" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/rest/reference/issues#timeline</a></p>
<p><a href="https://docs.github.com/en/developers/webhooks-and-events/events/issue-event-types#convert_to_draft" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/developers/webhooks-and-events/events/issue-event-types#convert_to_draft</a></p>
<p><a href="https://docs.github.com/en/developers/webhooks-and-events/events/issue-event-types#ready_for_review" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/developers/webhooks-and-events/events/issue-event-types#ready_for_review</a></p>

**Comment by StingyJack** [comment]
<p>Thanks, the timeline was exactly what I needed</p>

**Comment by XiaosongWen** [comment]
Hi I am wondering if there is a API to convert a draft pr to ready to review, I tried https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#update-a-pull-request . This seems not work

**Reply by StingyJack** [reply]
@XiaosongWen you should probably ask that in a separate question or discussion.
